### ISSUE TITLE
GMM notebook example: MCMC/NUTS simulation is not reproducible

### ISSUE THREAD

**Issue by inarighas** [pyro-ppl/numpyro#1616]
```
source: https://num.pyro.ai/en/stable/tutorials/gmm.html#MCMC
numpyro.__version__: 0.12.1
jax.__version__: 0.4.13
```

-- 

When running the collapsed NUTS to explore the full posterior, the obtained results did not match the presented ones.

```python
from numpyro.infer import MCMC, NUTS


kernel = NUTS(model)
mcmc = MCMC(kernel, num_warmup=50, num_samples=250)
mcmc.run(random.PRNGKey(2), data)
mcmc.print_summary()
posterior_samples = mcmc.get_samples()
```

Obtained posterior density:
![image](https://github.com/pyro-ppl/numpyro/assets/14874638/6bf8d1c2-3c8e-4004-b91e-757faf2472f6)

But, with longer `num_warmup` (to 150 or more),  we get roughly the expected behaviour:
 
![image](https://github.com/pyro-ppl/numpyro/assets/14874638/3cb1903c-1400-4d3d-8d68-ab3fa547c865)

With more samples (~2500), the pattern is better:
![image](https://github.com/pyro-ppl/numpyro/assets/14874638/26463f3a-341e-4ae3-bfd3-a6064baa9329)

I would like to clarify that in my previous attempts, I used the same values and parameters for reproduction. When I ran the code on Google Drive, the results matched the ones mentioned in the documentation. However, when I ran the code on my laptop, there was a significant difference. I can provide you with more details regarding this issue. Considering the specified random seeds and the simplicity of the example, I find the difference to be quite substantial.

I would like also to thank all the contributors for this library !

**Comment by ordabayevy** [comment]
@fehiepsi can this be due to a newer version of jax?

**Comment by fehiepsi** [comment]
Yes, it's likely caused by numerical changes. It makes sense to use higher warmup and larger num_samples. Currently, rhat, n_eff in the checked-in version is pretty poor.
